src/nftables.c

```diff
@@ -0,0 +1,371 @@
+
+  nftnl_rule_set_u32(r, NFTNL_RULE_FAMILY, family);
+
+  /* TODO: handle is not supported in plugin yet */
+  if (handle != NULL) {
+    handle_num = atoll(handle);
+    nftnl_rule_set_u64(r, NFTNL_RULE_POSITION, handle_num);
+  }
+
+  return r;
+} /* setup_rule */
+
+static int parse_rule_udata_cb(const struct nftnl_udata *attr, void *data) {
+  unsigned char *value = nftnl_udata_get(attr);
+  uint8_t type = nftnl_udata_type(attr);
+  uint8_t len = nftnl_udata_len(attr);
+  const struct nftnl_udata **tb = data;
+
+  switch (type) {
+  case NFTNL_UDATA_RULE_COMMENT:
+    if (value[len - 1] != '\0')
+      return -1;
+    break;
+  default:
+    return 0;
+  }
+  tb[type] = attr;
+  return 0;
+} /* parse_rule_udata_cb */
+
+static char *nftnl_rule_get_comment(const struct nftnl_rule *nlr) {
+  const struct nftnl_udata *tb[NFTNL_UDATA_RULE_MAX + 1] = {};
+  const void *data;
+  uint32_t len;
+
+  if (!nftnl_rule_is_set(nlr, NFTNL_RULE_USERDATA))
+    return NULL;
+
+  data = nftnl_rule_get_data(nlr, NFTNL_RULE_USERDATA, &len);
+
+  if (nftnl_udata_parse(data, len, parse_rule_udata_cb, tb) < 0)
+    return NULL;
+
+  if (!tb[NFTNL_UDATA_RULE_COMMENT])
+    return NULL;
+
+  return strdup(nftnl_udata_get(tb[NFTNL_UDATA_RULE_COMMENT]));
+} /* nftnl_rule_get_comment */
+
+static void submit(const char *table, const char *chain, char *comment,
+                   uint64_t bts, uint64_t cnt) {
+  value_list_t vl = VALUE_LIST_INIT;
+  int status;
+
+  sstrncpy(vl.plugin, "nftables", sizeof(vl.plugin));
+
+  status = ssnprintf(vl.plugin_instance, sizeof(vl.plugin_instance), "%s-%s",
+                     table, chain);
+  if ((status < 1) || ((unsigned int)status >= sizeof(vl.plugin_instance)))
+    return;
+
+  sstrncpy(vl.type_instance, comment, sizeof(vl.type_instance));
+
+  sstrncpy(vl.type, "ipt_bytes", sizeof(vl.type));
+  vl.values = &(value_t){.derive = (derive_t)bts};
+  vl.values_len = 1;
+  plugin_dispatch_values(&vl);
+
+  sstrncpy(vl.type, "ipt_packets", sizeof(vl.type));
+  vl.values = &(value_t){.derive = (derive_t)cnt};
+  plugin_dispatch_values(&vl);
+} /* void submit */
+
+static int submit_cb(struct nftnl_expr *e, void *data) {
+  const char *name = nftnl_expr_get_str(e, NFTNL_EXPR_NAME);
+  uint64_t bytes, count;
+
+  struct nftnl_rule *r = (struct nftnl_rule *)data;
+  const char *table = nftnl_rule_get_str(r, NFTNL_RULE_TABLE);
+  const char *chain = nftnl_rule_get_str(r, NFTNL_RULE_CHAIN);
+  char *comment = nftnl_rule_get_comment(r);
+  /* const uint64_t handle = nftnl_rule_get_u64(r, NFTNL_RULE_HANDLE); */
+
+  if (strcmp(name, "counter") == 0) {
+    count = nftnl_expr_get_u64(e, NFTNL_EXPR_CTR_PACKETS);
+    bytes = nftnl_expr_get_u64(e, NFTNL_EXPR_CTR_BYTES);
+    plugin_log(LOG_NOTICE, "Table: %s | Chain: %s | Comment: %s", table, chain,
+               comment);
+    plugin_log(LOG_NOTICE, "Bytes: %lu | Packets: %lu", bytes, count);
+    submit(table, chain, comment, bytes, count);
+  }
+  free(comment);
```

Comment:
Variable `comment` will either be the comment extracted from a `struct nftnl_rule` or NULL if the rule does not have any comment. See `nftnl_rule_get_comment`. Calling free when ptr is NULL should be safe, but a check can be added if this is not wanted.